src/exobot.js

```diff
@@ -73,6 +78,96 @@ export class Exobot {
     }, this.log.critical);
   }
 
+  async initUsers () {
+    await this.databaseInitialized();
+    this.users = this.db.get('exobot-users').value();
+    if (this.users) {
+      return;
+    }
+    this.db.set('exobot-users', {botUsers: {}}).value();
+    this.users = this.db.get('exobot-users').value();
+  }
+
+  mergeUsers (destUser, srcUser)
+  {
+    if (destUser && srcUser) {
+      merge(destUser.roles, srcUser.roles);
+      Object.keys(srcUser.adapters).map((adapter) => {
+        this.users[adapter][srcUser.adapters[adapter].userId].botID = destUser.id;
+      });
+      merge(destUser.adapters, srcUser.adapters);
+      delete this.users.botUsers[srcUser.id];
+      this.db.write();
+      return 'Login complete';
+    }
+
+  }
+
+  getUserRoles (userId) {
+    const user = this.users.botUsers[userId];
+    let roles = [];
+
+    if (user) {
+      Object.keys(user.adapters).map( (adapter) => {
```

Comment:
no spaces between parens here and below: `( (adapter)` should be `((adapter)`. you also don't need parens when you only have a single argument: `(adapter =>`